web/packages/teleterm/src/ui/services/resources/resourcesService.ts

```diff
@@ -199,15 +215,22 @@ export type SearchResultDatabase = {
   resource: types.Database;
 };
 export type SearchResultKube = { kind: 'kube'; resource: types.Kube };
+export type SearchResultApp = {
+  kind: 'app';
+  resource: ReturnType<typeof makeApp>;
```

Comment:
Maybe let's define this type explicitly in `ClustersService` as `App`? Otherwise whenever there's some other function that uses this type, it'll have to repeat `ReturnType<typeof makeApp>`.

Overall probably an even better approach would be to handle this universally on the tsh client level or something so that a regular app never enters the rest of the app and we always operate on this type with `addrWithProtocol` instead. But I don't think it's worth pursuing that at the moment.